--- build-server.py     2021-07-02 14:44:36.876163933 +0200
+++ build-server.lgtm.py        2021-07-02 14:39:26.201733928 +0200
@@ -25,6 +25,7 @@
 import optparse
 import atexit
 import platform
+import distro

 ####################
 ### Global variables
@@ -785,10 +786,14 @@
     excludes_list = [ '--exclude=%s' % pattern for pattern in ignored_patterns ]
     excludes = ' '.join(excludes_list)

-    tar_cmd = 'tar czf %(tarball_name)s %(versioned_serverdir)s %(excludes)s' \
+    # tar will copy the content the directory python3.[7-9]/ to python3.6/
+    transform = '--transform=\'s,python3.[7-9]/,python3/,\''
+
+    tar_cmd = 'tar czf %(tarball_name)s %(transform)s %(versioned_serverdir)s %(excludes)s' \
               % dict(tarball_name=tarball_name,
                      versioned_serverdir=versioned_serverdir,
-                     excludes=excludes)
+                     excludes=excludes,
+                     transform=transform)

     if run(tar_cmd) < 0:
         error('failed to generate the tarball')
@@ -807,16 +812,19 @@
     version = conf[CONF_VERSION]
     arch = os.uname()[-1].replace('_', '-')
-    if 'arm' in platform.machine():
+    if 'arm' in platform.machine() or 'aarch64' in platform.machine():
-        arch = 'pi'
+        arch = platform.machine()
     elif arch != 'x86-64':
         arch = 'i386'

+    # determine linux distribution
+    distribution = distro.linux_distribution()[-1].replace(' ', '-')
+
     dbg = ''
     if conf[CONF_NO_STRIP]:
         dbg = '.dbg'

-    tarball_name = 'seafile-server_%(version)s_%(arch)s%(dbg)s.tar.gz' \
-                   % dict(version=version, arch=arch, dbg=dbg)
+    tarball_name = 'seafile-server-%(version)s-%(distribution)s-%(arch)s%(dbg)s.tar.gz' \
+                   % dict(version=version, distribution=distribution, arch=arch, dbg=dbg)
     dst_tarball = os.path.join(conf[CONF_OUTPUTDIR], tarball_name)

     # generate the tarball
